database/database.go

```diff
@@ -100,13 +103,27 @@ func GetConfigRelayerIDs(cfg *config.Config) []RelayerID {
 // Calculate all of the possible relayer keys for a given source blockchain
 func GetSourceBlockchainRelayerIDs(sourceBlockchain *config.SourceBlockchain) []RelayerID {
 	var ids []RelayerID
-	for _, dst := range sourceBlockchain.GetSupportedDestinations().List() {
-		ids = append(ids, RelayerID{
-			SourceBlockchainID:      sourceBlockchain.GetBlockchainID(),
-			DestinationBlockchainID: dst,
-			OriginSenderAddress:     common.Address{}, // TODO: populate with allowed sender/receiver addresses
-			DestinationAddress:      common.Address{},
-		})
+	srcAddresses := sourceBlockchain.GetAllowedOriginSenderAddresses()
+	// If no addresses are provided, use the zero address to construct the relayer ID
+	if len(srcAddresses) == 0 {
+		srcAddresses = append(srcAddresses, AllAllowedAddress)
+	}
+	for _, srcAddress := range srcAddresses {
+		for _, dst := range sourceBlockchain.SupportedDestinations {
+			dstAddresses := dst.GetAddresses()
+			// If no addresses are provided, use the zero address to construct the relayer ID
+			if len(dstAddresses) == 0 {
+				dstAddresses = append(dstAddresses, AllAllowedAddress)
+			}
+			for _, dstAddress := range dstAddresses {
+				ids = append(ids, RelayerID{
```

Comment:
I think it'd be better to have `NewRelayerID` that initializes here too

Reply:
I don't think there's any real benefit to adding a standalone constructor on a POD datatype. It would result in a net increase of code, since it would take the exact same arguments as the composite literal here.

Reply:
I was originally thinking to initialize the `RelayerID.id` field in the construtor instead of in `GetID`, but looking back not important to add here.

Reply:
Good point re: initializing the `id` field. I added a constructor, and made `ID` exported so that it can be referenced directly.